include/zephyr/arch/riscv/arch.h

```diff
@@ -43,12 +43,16 @@
  * configurable stack wiggle room to execute the fault handling code off of,
  * as well as some guard size to cover possible sudden stack pointer
  * displacement before the fault.
- *
- * The m-mode PMP set is not overly used so no need to force NAPOT.
  */
+#ifdef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
```

Comment:
This might want a new kconfig.  The original is designed to reflect limitations of the hardware, and is set by the platform and invariant with respect to app choices.

This is here just to make a tradeoff decision between preferrentially wasting MPU entries or main memory bytes, which is an optimization thing and absolutely the app's business.

Reply:
I,m not sure I understand the nuance... but there you have it.